zerver/migrations/0055_move_avatars_to_be_uid_based.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.5 on 2017-02-27 17:03
+from __future__ import unicode_literals
+
+from django.db import migrations
+from django.db.backends.postgresql_psycopg2.schema import DatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.core.files.uploadedfile import SimpleUploadedFile
+from zerver.lib.upload import upload_avatar_image
+from zerver.lib.avatar import get_avatar_url
+import requests
+
+def move_avatars_to_be_uid_based(apps, schema_editor):
+    # type: (StateApps, DatabaseSchemaEditor) -> None
+    user_profile_model = apps.get_model('zerver', 'UserProfile')
+    for user_profile in user_profile_model.objects.filter(avatar_source=u"U"):
+        url = get_avatar_url(user_profile.avatar_source,
+                             user_profile.email,
+                             user_profile.avatar_version)
+        avatar_connection = requests.get(url)
+        if avatar_connection.status_code == 200:
+            avatar_data = avatar_connection.content
+            avatar_file = SimpleUploadedFile('avatar.jpg', avatar_data, 'image/jpeg')
+            upload_avatar_image(avatar_file, user_profile, user_profile.id)
```

Comment:
I would prefer to do something that preserves any metadata on the image object, rather than doing a full fetch+re-upload, if it's reasonable to do that...